doc/concepts/configuration/configuration_persistence.rst

```diff
@@ -0,0 +1,233 @@
+
+..  _configuration_persistence_wal_mode:
+
+Set the WAL mode
+~~~~~~~~~~~~~~~~
+
+To be able to recover data in case of a possible instance restart, enable recording to the write-ahead log.
+To do it, set the :ref:`wal.mode <configuration_reference_wal_mode>` configuration option to ``write`` or ``fsync``.
+The example below shows how to specify the ``write`` WAL mode for ``instance001``:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/persistence_wal/config.yaml
+    :language: yaml
+    :start-at: instance001
+    :end-at: 'write'
+    :dedent:
+
+The ``write`` mode enables WAL and writes the data without waiting the data to be flushed to the storage device.
+The ``fsync`` mode enables WAL and ensures that the record is written to the storage device.
+
+To turn the WAL writer off, set the ``wal.mode`` option to ``none``.
+
+..  _configuration_persistence_wal_dir:
+
+Specify a directory for WAL files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To configure a directory where the WAL files are stored, use the :ref:`wal.dir <configuration_reference_wal_dir>` configuration option.
+The example below shows how to specify a directory for ``instance001`` explicitly:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/persistence_wal/config.yaml
+    :language: yaml
+    :start-at: wal:
+    :end-at: 'var/lib/{{ instance_name }}/wals'
+    :dedent:
+
+
+..  _configuration_persistence_wal_rescan:
+
+Set an interval between scans
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If case of :ref:`replication <replication>` or :ref:`hot standby <index-hot_standby>` mode,
+Tarantool scans for changes in the WAL files every :ref:`wal.dir_rescan_delay <configuration_reference_wal_dir_rescan_delay>`
+seconds. The example below shows how to specify the interval between scans:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/persistence_wal/config.yaml
+    :language: yaml
+    :start-at: dir_rescan_delay
+    :end-before: cleanup_delay
+    :dedent:
+
+..  _configuration_persistence_wal_maxsize:
+
+Set a maximum size for the WAL file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A new WAL file is created when the current one reaches the :ref:`wal.max_size <configuration_reference_wal_max_size>`
+size. The configuration for this option might look as follows:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/persistence_wal/config.yaml
+    :language: yaml
+    :start-at: max_size
+    :end-at: 268435456
+    :dedent:
+
+..  _configuration_persistence_wal_rescan:
+
+Set a delay for the garbage collector
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Tarantool, the :ref:`checkpoint daemon <configuration_persistence_checkpoint_daemon>`
+takes new snapshots at the given interval (see :ref:`snapshot.by.interval <configuration_reference_snapshot_by_interval>`).
+After an instance restart, the daemon activates the Tarantool garbage collector that deletes the old WAL files.
```

Comment:
Why we describe checkpoint daemon in the garbage collector section and vice versa? It feels as mixed together without a reason. Can we split these topic to make each of them more clear?

One subsystem creates a schedule when to perform a new snapshot based on the configuration options and on how fast the WAL files are grow. And makes these snapshots.

Another subsystems tracks which snapshots are needed by relays and other consumers and deletes unnecessary ones.

Reply:
Thanks for noticing! I split this description into separate sections - Checkpoint daemon and Tarantool garbage collector